porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -105,24 +115,23 @@ func (r *PackageVariantReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	if errs := validatePackageVariant(pv); len(errs) > 0 {
-		pv.Status.ValidationErrors = nil
-		for _, validationErr := range errs {
-			if validationErr.Error() != "" {
-				pv.Status.ValidationErrors = append(pv.Status.ValidationErrors, validationErr.Error())
-			}
-		}
-		statusUpdateErr := r.Client.Status().Update(ctx, pv)
-		return ctrl.Result{}, statusUpdateErr
+		setValidationConditionsToFalse(pv, combineErrors(errs))
+		return ctrl.Result{}, nil
 	}
-
-	upstream := r.getUpstreamPR(pv.Spec.Upstream, prList)
-	if upstream == nil {
-		return ctrl.Result{}, fmt.Errorf("could not find upstream package revision")
+	upstream, err := r.getUpstreamPR(pv.Spec.Upstream, prList)
+	if err != nil {
+		setValidationConditionsToFalse(pv, err.Error())
+		return ctrl.Result{}, nil
```

Comment:
As stated above I think we should set this to requeue after a short time in this instance. And if it's easy we should implement backoff on that requeueing. If that part is not easy, it can wait but we probably should open a ticket to reminde us.

Reply:
I can't find official documentation, but from these comments:

- https://github.com/kubernetes-sigs/controller-runtime/issues/808#issuecomment-639845414
- https://github.com/crossplane/crossplane-runtime/issues/40#issuecomment-777145647

it looks like requeueing with exponential backoff is as easy as just returning an error here. So I updated this to return an error.